frontend/src/components/Login.vue

```diff
@@ -63,46 +25,41 @@ import {HRequest} from '../utils/HRequests'
 export default {
   name: 'Login',
   components: {HFooter},
-  beforeCreate () {
-    this.form = this.$form.createForm(this)
+  mounted () {
+    this.getProviders()
+  },
+  data () {
+    return {
+      openidProviders: []
+    }
   },
   methods: {
-    handleSubmit (e) {
-      e.preventDefault()
-      this.form.validateFields((err, values) => {
-        // ``values`` stands for the form data
-        if (!err) {
-          const qs = require('qs')
-          let message = this.$message
-          const options = {
-            method: 'POST',
-            headers: { 'content-type': 'application/x-www-form-urlencoded' },
-            data: qs.stringify({
-              user: values.username,
-              password: values.password
-            }),
-            url: '/api/auth/login'}
-          HRequest(options).then(
-            (response) => {
-              // the initial ``data`` is data in response object
-              // second ``data`` is the data in message object
-              if (response.data.data.success) {
-                this.onSuccess()
-              } else {
-                message.warning(response.data.data.msg)
-              }
-            }
-          )
+    getProviders () {
+      HRequest.get('/api/auth/providers').then(
+        (response) => {
+          this.openidProviders = response.data.data
         }
-      })
+      ).catch()
     },
-    onSuccess () {
-      let next = this.$route.query.next
-      if (next) {
-        this.$router.push(next)
-      } else {
-        this.$router.push({name: 'Home'})
-      }
+    onLogin () {
+      console.log()
+      let provider = event.target.name
+      let strWindowFeatures = 'toolbar=no, menubar=no, width=800, height=600, top=100, left=100'
+      const popup = window.open(`/auth/oauth/${provider}`, 'oauth', strWindowFeatures)
+      if (!popup) return 'POPUP_FAILED'
+      popup.focus()
+      const app = this
+      var timer = setInterval(function () {
```

Comment:
用箭头函数就不会有取不到 this 的问题了 
```suggestion
      var timer = setInterval( () => {
```